components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/IoTHubMessageSink.cs

```diff
@@ -62,15 +62,20 @@ public class IoTHubMessageSink : IMessageSink, IDisposable {
                     kMessagesSent.WithLabels(IotHubMessageSinkGuid, IotHubMessageSinkStartTime).Set(SentMessagesCount);
                     SentMessagesCount = 0;
                 }
-                using(kSendingDuration.NewTimer()) {
+                using (kSendingDuration.NewTimer()) {
                     var sw = new Stopwatch();
                     sw.Start();
 
-                    if (messagesCount == 1) {
-                        await _clientAccessor.Client.SendEventAsync(messageObjects.First()).ConfigureAwait(false);
+                    try {
+                        if (messagesCount == 1) {
+                            await _clientAccessor.Client.SendEventAsync(messageObjects.First()).ConfigureAwait(false);
+                        }
+                        else {
+                            await _clientAccessor.Client.SendEventBatchAsync(messageObjects).ConfigureAwait(false);
+                        }
                     }
-                    else {
-                        await _clientAccessor.Client.SendEventBatchAsync(messageObjects).ConfigureAwait(false);
+                    catch (Exception e) {
+                        _logger.Error(e, "Error sending message(s) to IoT Hub");
```

Comment:
Would be good to show failure to send messages in diganostics maybe...

Reply:
Created a backlog item for this